src/lib/markbind/src/parser.js

```diff
@@ -18,888 +19,738 @@ _.pick = require('lodash/pick');
 const CyclicReferenceError = require('./handlers/cyclicReferenceError.js');
 const md = require('./lib/markdown-it');
 const utils = require('./utils');
+const parserUtils = require('./parserUtils');
 
 cheerio.prototype.options.xmlMode = true; // Enable xml mode for self-closing tag
 cheerio.prototype.options.decodeEntities = false; // Don't escape HTML entities
 
 const {
   ATTRIB_INCLUDE_PATH,
   ATTRIB_CWF,
-  BOILERPLATE_FOLDER_NAME,
   IMPORTED_VARIABLE_PREFIX,
 } = require('./constants');
 
-const VARIABLE_LOOKUP = new Map();
-const FILE_ALIASES = new Map();
-const PROCESSED_INNER_VARIABLES = new Set();
-
-/*
- * Utils
- */
-
-/**
-* @throws Will throw an error if a non-absolute path or path outside the root is given
-*/
-function calculateNewBaseUrls(filePath, root, lookUp) {
-  if (!path.isAbsolute(filePath)) {
-    throw new Error(`Non-absolute path given to calculateNewBaseUrls: '${filePath}'`);
-  }
-  if (!pathIsInside(filePath, root)) {
-    throw new Error(`Path given '${filePath}' is not in root '${root}'`);
+// const VARIABLE_LOOKUP = new Map();
+// const FILE_ALIASES = new Map();
+// const PROCESSED_INNER_VARIABLES = new Set();
+class Parser {
+  constructor(options) {
+    this._options = options || {};
+    // eslint-disable-next-line no-console
+    this._onError = this._options.errorHandler || console.error;
+    this._fileCache = {};
+    this.dynamicIncludeSrc = [];
+    this.staticIncludeSrc = [];
+    this.boilerplateIncludeSrc = [];
+    this.missingIncludeSrc = [];
+    this.VARIABLE_LOOKUP = new Map();
+    this.FILE_ALIASES = new Map();
+    this.PROCESSED_INNER_VARIABLES = new Set();
   }
-  function calculate(file, result) {
-    if (file === root) {
-      return { relative: path.relative(root, root), parent: root };
-    }
-    const parent = path.dirname(file);
-    if (lookUp.has(parent) && result.length === 1) {
-      return { relative: path.relative(parent, result[0]), parent };
-    } else if (lookUp.has(parent)) {
-      return calculate(parent, [parent]);
-    }
-    return calculate(parent, result);
-  }
-
-  return calculate(filePath, []);
-}
-
-function calculateBoilerplateFilePath(pathInBoilerplates, asIfAt, config) {
-  const { parent, relative } = calculateNewBaseUrls(asIfAt, config.rootPath, config.baseUrlMap);
-  return path.resolve(parent, relative, BOILERPLATE_FOLDER_NAME, pathInBoilerplates);
-}
```

Comment:
Are these methods that you moved used by anything else other than `Parser`? It is OK to leave them inside parser if no one else is using them, rather than refactoring into another file such as `parserUtils.js`.